File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
let [A, B] = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split(" ")
  .map(Number);

function gcd(a, b) {
  while (b !== 0) {
    let temp = a % b;
    a = b;
    b = temp;
  }
  return a;
}

function lcm(a, b) {
  return a * (b / gcd(a, b));
}

console.log(gcd(A, B));
console.log(lcm(A, B));

File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.kt
fun main() {
    val (N, M) = readln().split(" ").map { it.toInt() }
    println(gcd(N, M))
    println(lcm(N, M))
}

fun gcd(a: Int, b: Int): Int {
    var a = a
    var b = b
    while (b != 0) {
        val temp = a % b
        a = b
        b = temp
    }
    return a
}

fun lcm(a: Int, b: Int): Int {
    return a * (b / gcd(a, b))
}